include/deal.II/non_matching/mapping_info.h

```diff
@@ -1170,9 +1190,19 @@ namespace NonMatching
                     inverse_jacobians[offset][s][d], v) =
                     mapping_data.inverse_jacobians[q * n_lanes + v][s][d];
             if (update_flags_mapping & UpdateFlags::update_JxW_values)
-              dealii::internal::VectorizedArrayTrait<Number>::get(
-                JxW_values[offset], v) =
-                mapping_data.JxW_values[q * n_lanes + v];
+              {
+                if (!quadrature_weights_are_JxW)
```

Comment:
Can you change the logic here to positive to make this easier to read?

Reply:
[33b8142](https://github.com/dealii/dealii/pull/15292/commits/33b8142c76dc29c208fbb821937e3efa5077e065)